pipeline/matching/generators/similar_intents.py

```diff
@@ -15,4 +25,65 @@ def __init__(self, min_common: int = 0):
 
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
         # TODO: Implement your generator
-        yield from []
+        # inp.logger.info("Lauren")
+
+        users = inp.users
+        # print(users)
+        match = False
+
+        match_list = []
+
+        # iterate over the arr of users
+        for idx, u in enumerate(users):
+            # take Intent info from the current user
+            intents = u.intents
+            intent = intents[0]
+            code = intent.code
+            side = intent.side
+            name = intent.name
+
+            # o therSide would be the other side of the intent that they have
+            if side == Side.GIVING:
+                otherSide = Side.SEEKING
+            elif side == Side.SEEKING:
+                otherSide = Side.GIVING
+            else:
+                otherSide = Side.BLANK
```

Comment:
This is a good idea! Makes it easier to find the compatible intent.

Two thoughts to make this code simpler:

1. You can assume that the intent side will be either GIVING or SEEKING. If it is not, that is a problem that occurs before your code. If you want, you could raise a ValueError if you encounter a side that is not one of those values, but this is optional.

2. Since we can assume the side will be either SEEKING or GIVING, you could simplify this block of code with a [ternary statement](https://book.pythontips.com/en/latest/ternary_operators.html) (like the ones you used below):

```python
other_side = Side.GIVING if side == Side.SEEKING else Side.SEEKING
```

Reply:
> This is a good idea! Makes it easier to find the compatible intent.
> 
> Two thoughts to make this code simpler:
> 
> 1. You can assume that the intent side will be either GIVING or SEEKING. If it is not, that is a problem that occurs before your code. If you want, you could raise a ValueError if you encounter a side that is not one of those values, but this is optional.

If I raise a ValueError, should this be something the Ranking Team might need to know for their implementation? Also, would it stop the generator from returning anything?

> 2. Since we can assume the side will be either SEEKING or GIVING, you could simplify this block of code with a [ternary statement](https://book.pythontips.com/en/latest/ternary_operators.html) (like the ones you used below):
> 
> ```python
> other_side = Side.GIVING if side == Side.SEEKING else Side.SEEKING
> ```
Thanks for the suggestion!

Reply:
> 2. Since we can assume the side will be either SEEKING or GIVING, you could simplify this block of code with a [ternary statement](https://book.pythontips.com/en/latest/ternary_operators.html) (like the ones you used below):
> 
> ```python
> other_side = Side.GIVING if side == Side.SEEKING else Side.SEEKING
> ```

Resolved on line 47 of similar_intents.py